src/main/java/org/terasology/scenario/components/information/ConcatStringComponent.java

```diff
@@ -18,6 +18,9 @@
 import org.terasology.entitySystem.Component;
 import org.terasology.network.Replicate;
 
+/**
+ * Component that denotes two strings to be concatenated

```

Comment:
I think a comment like this might be better:

"Marks a string expression entity (see {@link ScenarioStringComponent}" as being a string expression that conctenates strings. Entities with this component shoudl also have a {@link ArgumentContainerComponent} with the arguments string1 and string2 that specify the string expressions to concat."